### PR TITLE
Update referencing secrets document

### LINKED ISSUE #699: Update secret-reader to dapr-secret-reader under non-default namespace guidance
## Describe the proposal
When following the description in https://github.com/dapr/docs/blob/master/concepts/secrets/component-secrets.md, one will end up with a default service account with read access to all secrets in the current namespace.

This is not needed as the Component only needs read access to specific secrets. It is also problematic as all (dapr enabled) pods in the same namespace will have access to the same secrets.

I propose the following change to the documentation

from:
<pre>
apiVersion: rbac.authorization.k8s.io/v1
kind: Role
metadata:
  name: secret-reader
  namespace: <NAMESPACE>
rules:
- apiGroups: [""]
  resources: ["secrets"]
  verbs: ["get"]
</pre>

to:
<pre>
apiVersion: rbac.authorization.k8s.io/v1
kind: Role
metadata:
  name: secret-reader
  namespace: <NAMESPACE>
rules:
- apiGroups: [""]
  resources: ["secrets"]
  resourceNames: ["secret-name-1", "secret-name-2"]
  verbs: ["get"]
</pre>

And probably a line  telling that the sample secret names should be substituted by actual secret names. It might also be an option to rename toe role from "secret-reader" to "dapr-secret-reader" as it will now only give read access to Dapr specific secrets.

This will reduce the attack surface.

I assume it's also an option to advice on creating dedicated service accounts when running more than one pod in the application namespace. The Daprd sidecar seems to work well also when not using the default service account (only slightly tested).

### LINKED ISSUE #1034: Referencing secrets doc needs to be generalized and more detailed
**What content needs to be created or modified?**
Information on referencing secrets in components needs to be updated to be generalized and more clear.

https://docs.dapr.io/operations/components/component-secrets/

**Describe the solution you'd like**
Better instructions and examples

**Where should the new material be placed?**
https://docs.dapr.io/operations/components/component-secrets/

### PR DESCRIPTION
**Pull request by mukundansundar**
Thank you for helping make the Dapr documentation better!

If you are a new contributor, please see the [this contribution guidance](https://docs.dapr.io/contributing/contributing-docs/) which helps keep the Dapr documentation readable, consistent and useful for Dapr users.

Note that you must see that the suggested changes do not break the website [docs.dapr.io](https://docs.dapr.io). See [this overview](https://github.com/dapr/docs/blob/master/README.md#overview) on how to setup a local version of the website and make sure the website is built correctly.

In addition, please fill out the following to help reviewers understand this pull request:

## Description

Updated referencing secrets documentation.

## Issue reference

_Please reference the issue this PR will close: #1034

### CODE REVIEW

File: daprdocs/content/en/operations/components/component-secrets.md [hunk 1]
```diff
@@ -88,12 +57,14 @@ spec:
...
+  secretStore: <SECRET_STORE_NAME>
 ```
 
-The above example tells Dapr to use the `kubernetes` secret store, extract a secret named `redis-secret` and assign the value of the `redis-password` key in the secret to the `redisPassword` field in the Component.
+When running in Kubernetes and using a Kubernetes secret store, either the field `auth.SecretStore` can be empty (as it is assumed to be Kubernetes secret store) or it needs to be `kubernetes`. For all other secret store, the `SECRET_STORE_NAME` is the name of the configured secret store component.
+
+The above example tells Dapr to extract a secret named `redis-secret` from the defined secret store and assign the value of the `redis-password` key in the secret to the `redisPassword` field in the Component.
 
-### Creating a secret and referencing it in a Component
+### Creating a Kubernetes secret and referencing it in a Component
```
Comment (AaronCrawfis) [inline daprdocs/content/en/operations/components/component-secrets.md]: Component should be lowercase

File: daprdocs/content/en/operations/components/component-secrets.md [hunk 2]
```diff
@@ -18,40 +18,9 @@ When running in Kubernetes, if the `auth.secretStore` is empty, the Kubernetes s
...
-roleRef:
-  kind: Role
-  name: secret-reader
-  apiGroup: rbac.authorization.k8s.io
-```
-
 ## Examples
 
-Using plain text:
+Using plain text secrets (not recommended for production):
```
Comment (AaronCrawfis) [inline daprdocs/content/en/operations/components/component-secrets.md]: This should be a ### header, and move `(not recommended for production)` into the next line as a warning:

```html
{{% alert title="Note" color="warning" %}}
Plaintext secrets are not recommended for production environments. Use a secret store instead to securly store and access keys.
{{% /alert %}}
```

**Review by yaron2 — APPROVED** [review]
lgtm

File: daprdocs/content/en/operations/components/component-secrets.md [hunk 3]
```diff
@@ -88,12 +57,14 @@ spec:
     	name: redis-secret
         key:  redis-password
 auth:
-  secretStore: kubernetes
+  secretStore: <SECRET_STORE_NAME>
 ```
 
-The above example tells Dapr to use the `kubernetes` secret store, extract a secret named `redis-secret` and assign the value of the `redis-password` key in the secret to the `redisPassword` field in the Component.
+When running in Kubernetes and using a Kubernetes secret store, either the field `auth.SecretStore` can be empty (as it is assumed to be Kubernetes secret store) or it needs to be `kubernetes`. For all other secret store, the `SECRET_STORE_NAME` is the name of the configured secret store component.
```
Comment (yaron2) [inline daprdocs/content/en/operations/components/component-secrets.md]: change to `auth.secretStore`

File: daprdocs/content/en/operations/components/component-secrets.md (hunk 1 above)
Comment (mukundansundar) [reply daprdocs/content/en/operations/components/component-secrets.md]: This is not a component. It is actually referring to the k8s platform.

File: daprdocs/content/en/operations/components/component-secrets.md [hunk 4]
```diff
@@ -69,7 +38,7 @@ spec:
     value: MyPassword
 ```
 
-Using a Kubernetes secret:
+Referencing secret from a secret store:
```
Comment (AaronCrawfis) [inline daprdocs/content/en/operations/components/component-secrets.md]: Should be a ### header

File: daprdocs/content/en/operations/components/component-secrets.md (hunk 3 above)
Comment (AaronCrawfis) [inline daprdocs/content/en/operations/components/component-secrets.md]: Flip the order so it reads: 

`SECRET_STORE_NAME` is the name of the configured [secret store component]({{< ref supported-secret-stores >}}).
When running in Kubernetes and using a Kubernetes secret store, the field `auth.SecretStore` defaults to `kubernetes` and can be left empty.

**Review by AaronCrawfis — APPROVED** [review]

**Pull request merged**